Use the document base url when resolving script URLs

liamsi opened the pull request:
Does this resolve #10578?

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/servo/servo/10591)

<!-- Reviewable:end -->

liamsi committed:
Use the document base url when resolving script URLs

highfive commented:
Thanks for the pull request, and welcome! The Servo team is excited to review your changes, and you should hear from @simonsapin (or someone else) soon.

highfive commented:
Heads up! This PR modifies the following files:
 * @kichjang: components/script/dom/htmlscriptelement.rs

KiChjang commented on tests/wpt/metadata/html/semantics/scripting-1/the-script-element/fetch-src/alpha/base.html.ini:
```diff
@@ -1,5 +1,5 @@
   type: testharness
   [Script src with a base URL]
-    expected: FAIL
+    expected: PASS
```
No need to explicitly mark it as PASS; if the test passes, you can just remove the `expected: ` key-value pair. Since there's only one expected failure in this file previously, the entire file can be removed as well.

KiChjang commented on components/script/dom/htmlscriptelement.rs:
```diff
@@ -274,7 +274,8 @@ impl HTMLScriptElement {
         let window = window.r();
-        let base_url = window.get_url();
+        let document = document_from_node(self);
+        let base_url = document.base_url();
```
Is `document` being used anywhere else afterwards? If not, we can just use a one-liner (i.e. `let base_url = document_from_node(self).base_url();`)

Liamsi replied:
It actually is used (some lines above). There is a variable called `doc`. I'll work on your (and @Ms2ger) comments later (after work).

Ms2ger commented on components/script/dom/htmlscriptelement.rs:
```diff
@@ -274,7 +274,8 @@ impl HTMLScriptElement {
 
         let window = window_from_node(self);
         let window = window.r();
```
`window` is now only used for the `networking_task_source` call below. Please remove these two variables and use `document.window()` below.

KiChjang commented:
This looks pretty neat! Thanks for working on this!

@bors-servo r+

bors-servo commented:
:pushpin: Commit 209d336 has been approved by `KiChjang`

<!-- @bors-servo r=KiChjang 209d3363091300f7f5da8b4b826fc41ed8da3b80 -->

bors-servo commented:
:hourglass: Testing commit 209d336 with merge 9948ada...

bors-servo commented:
:broken_heart: Test failed - [linux-rel](http://build.servo.org/builders/linux-rel/builds/330)

KiChjang commented:
I think you just did a `git merge` on this PR - that's a big no-no. You should have rebased instead.

Liamsi commented:
> I think you just did a `git merge` on this PR - that's a big no-no. You should have rebased instead. Also, needs squash.

Yepp, sorry for that and thanks for the quick feedback. Rebased, squashed/fixed commits (and force pushed) my branch.

KiChjang commented:
@bors-servo r+

Thanks!

bors-servo commented:
:pushpin: Commit 62182f9 has been approved by `KiChjang`

<!-- @bors-servo r=KiChjang 62182f9de9e3ab20b439cb9750e377df0b06b500 -->

bors-servo commented:
:hourglass: Testing commit 62182f9 with merge 9896e53...

bors-servo commented:
:sunny: Test successful - [android](http://build.servo.org/builders/android/builds/287), [arm32](http://build.servo.org/builders/arm32/builds/307), [arm64](http://build.servo.org/builders/arm64/builds/299), [linux-dev](http://build.servo.org/builders/linux-dev/builds/305), [linux-rel](http://build.servo.org/builders/linux-rel/builds/335), [mac-dev-unit](http://build.servo.org/builders/mac-dev-unit/builds/310), [mac-rel-css](http://build.servo.org/builders/mac-rel-css/builds/302), [mac-rel-wpt](http://build.servo.org/builders/mac-rel-wpt/builds/307)
